packages/yew/src/dom_bundle/dom_position.rs

```diff
@@ -0,0 +1,254 @@
+        Self {
+            variant: DomSlotVariant::Node(next_sibling),
+        }
+    }
+
+    /// A new "placeholder" [DomSlot] that should not be used to insert nodes
+    #[inline]
+    pub fn new_debug_trapped() -> Self {
+        #[cfg(debug_assertions)]
+        {
+            Self::at(TRAP.with(|trap| trap.clone()))
+        }
+        #[cfg(not(debug_assertions))]
+        {
+            Self::at_end()
+        }
+    }
+
+    /// Get the [Node] that comes just after the position, or `None` if this denotes the position at
+    /// the end
+    fn with_next_sibling<R>(&self, f: impl FnOnce(Option<&Node>) -> R) -> R {
+        let checkedf = |node: Option<&Node>| {
+            #[cfg(debug_assertions)]
+            TRAP.with(|trap| {
+                assert!(
+                    node != Some(trap),
+                    "Should not use a trapped DomSlot. Please report this as an internal bug in \
+                     yew."
+                )
+            });
+            f(node)
+        };
+
+        match &self.variant {
+            DomSlotVariant::Node(ref n) => checkedf(n.as_ref()),
+            DomSlotVariant::Chained(ref chain) => chain.with_next_sibling(checkedf),
+        }
+    }
+
+    /// Insert a [Node] at the position denoted by this slot. `parent` must be the actual parent
+    /// element of the children that this slot is implicitly a part of.
+    pub(super) fn insert(&self, parent: &Element, node: &Node) {
+        self.with_next_sibling(|next_sibling| {
+            parent
+                .insert_before(node, next_sibling)
+                .unwrap_or_else(|err| {
+                    let msg = if next_sibling.is_some() {
+                        "failed to insert node before next sibling"
+                    } else {
+                        "failed to append child"
+                    };
+                    // Log normally, so we can inspect the nodes in console
+                    gloo::console::error!(msg, err, parent, next_sibling, node);
+                    // Log via tracing for consistency
+                    tracing::error!(msg);
+                    // Panic to short-curcuit and fail
+                    panic!("{}", msg)
+                });
+        });
+    }
+
+    #[cfg(target_arch = "wasm32")]
+    #[cfg(test)]
+    fn get(&self) -> Option<Node> {
+        self.with_next_sibling(|n| n.cloned())
+    }
+}
+
+impl DynamicDomSlot {
+    /// Create a dynamic dom slot that initially represents ("targets") the same slot as the
+    /// argument.
+    pub fn new(initial_position: DomSlot) -> Self {
+        Self {
+            target: Rc::new(RefCell::new(initial_position)),
+        }
+    }
+
+    pub fn new_debug_trapped() -> Self {
+        Self::new(DomSlot::new_debug_trapped())
+    }
+
+    /// Change the [`DomSlot`] that is targeted. Subsequently, this will behave as if `self` was
+    /// created from the passed DomSlot in the first place.
+    pub fn retarget(&self, next_position: DomSlot) {
```

Comment:
If we don't call it `RetargetableDomSlot` any more, should we also give another name than retarget?

Reply:
Changed to `reassign` which seems to be better fitting.